StudentAppMvc/StudentAppMvc/Controllers/ThtrStudentController.cs

```diff
@@ -0,0 +1,189 @@
+        /// <returns></returns>
+        public ActionResult Search(string searchName = "", string searchGender = "")
+        {
+            StudentListViewModel studentListDTO = new StudentListViewModel(_studentService.Search(searchName, searchGender), searchName: searchName, searchGender: searchGender);
+            return View(studentListDTO);
+        }
+
+
+        // GET: StudentController/Details/5
+        public ActionResult Details(int id)
+        {
+            Student student = _studentService.GetById(id);
+            ViewBag.Student = student;
+            return View(student);
+        }
+
+
+        // GET: StudentController/Create (Get View)
+        public ActionResult Create()
+        {
+            return View();
+        }
+
+
+        // POST: StudentController/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Create(Student student)
+        {
+            try
+            {
+                var result = _validationService.Validate(student);
+                if (result.Any())
+                {
+                    throw new Exception("error");
+                }
+                _studentService.Create(student);
+                return RedirectToAction(nameof(Index));
+            }
+            catch (DuplicateObjectException doe)
+            {
+                ModelState.AddModelError("Email", doe.Message);
+                return View(student);
+            }
+            catch (Exception)
+            {
+                return View();
+            }
+        }
+
+        //Get view student
+        public ActionResult Edit(int id)
+        {
+            Student student = _studentService.GetById(id);
+
+            return View(student);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        //public ActionResult Edit(int id, Student student)
+        public ActionResult Edit(Student student)
+        {
+            try
+            {
+                _studentService.Edit(student);
+            }
+            catch (DuplicateObjectException doe)
+            {
+                ModelState.AddModelError("Email", doe.Message);
+                return View(student);
+            }
+            catch (Exception)
+            {
+                return View(student);
+            }
+
+            return RedirectToAction(nameof(Index));
+            
+        }
+
+        //Delete student by studentId
+        public ActionResult Delete(int id)
+        {
+            try 
+            {
+                _studentService.Delete(id);
+            }
+            catch(ObjectExistInDTOException oe)
+            {
+                Console.WriteLine(oe.Message);
+
+            }catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }
+
+            return RedirectToAction(nameof(Index));
+        }
+
+        //[ValidateAntiForgeryToken]
+        public ActionResult CreateMark(int id)
```

Comment:
Perhaps, could be a better idea to move Mark to MarkController. And we can use the word Grade instead. Mark has a lot of meanings